File: frontend/src/sentry.ts
/**
 * Sentry initialization for React frontend.
 *
 * If VITE_SENTRY_DSN is not set, Sentry is not initialized (no-op).
 * This allows development without a Sentry account.
 */

import * as Sentry from "@sentry/react";

const SENTRY_DSN = import.meta.env.VITE_SENTRY_DSN as string | undefined;

export function initSentry(): boolean {
  if (!SENTRY_DSN) {
    // eslint-disable-next-line no-console
    console.info("[Sentry] DSN not set — error monitoring disabled");
    return false;
  }

  Sentry.init({
    dsn: SENTRY_DSN,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.replayIntegration({
        maskAllText: false,
        blockAllMedia: false,
      }),
    ],
    // Performance Monitoring: capture 10% of transactions
    tracesSampleRate: 0.1,
    // Session Replay: capture 10% of sessions
    replaysSessionSampleRate: 0.1,
    // Session Replay: capture 100% of sessions with errors
    replaysOnErrorSampleRate: 1.0,
  });

  // eslint-disable-next-line no-console
  console.info("[Sentry] Initialized successfully");
  return true;
}

File: backend/app/core/sentry.py
"""Sentry SDK initialization for error monitoring.

If SENTRY_DSN is not set, Sentry is not initialized (no-op).
This allows development without a Sentry account.
"""

import logging

from app.config import settings

logger = logging.getLogger(__name__)


def init_sentry() -> bool:
    """Initialize Sentry SDK if a DSN is configured.

    Returns:
        True if Sentry was initialized, False otherwise.
    """
    if not settings.SENTRY_DSN:
        logger.info("Sentry DSN not set — error monitoring disabled")
        return False

    import sentry_sdk
    from sentry_sdk.integrations.fastapi import FastApiIntegration
    from sentry_sdk.integrations.starlette import StarletteIntegration

    sentry_sdk.init(
        dsn=settings.SENTRY_DSN,
        traces_sample_rate=settings.SENTRY_TRACES_SAMPLE_RATE,
        integrations=[
            StarletteIntegration(transaction_style="endpoint"),
            FastApiIntegration(transaction_style="endpoint"),
        ],
        environment="production" if not settings.DEBUG else "development",
        # Send PII (email, username) for better error context
        send_default_pii=True,
    )

    logger.info("Sentry initialized successfully (env=%s)", "production" if not settings.DEBUG else "development")
    return True
